src/lib/components/composites/search-bar/SearchBar.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+    import { Input } from "$lib/components/primitives/input/index";
+    import { Search } from "lucide-svelte";
+
+    interface Props {
+        placeholderText?: string;
+        onSearch: (searchText: string) => void;
+    }
+    const { placeholderText = "Search", onSearch }: Props = $props();
+
+    let searchInput: string = $state("");
+
+    // timeout in ms to wait after last key input before start search
+    const timeout: number = 500;
+    let timeoutId: ReturnType<typeof setTimeout> | null = null;
+
+    const processNewInput = (event: KeyboardEvent) => {
+        if (event.key == "Escape") {
+            // Check, whether user pressed the 'Esc' character to clear input and dispatch focus
+            searchInput = "";
+            if (timeoutId !== null) clearTimeout(timeoutId);
+
+            (document.activeElement as HTMLInputElement).blur();
+        } else if (event.key == "Enter" && searchInput !== "") {
+            // Check, whether user pressed 'Enter' to start the search directly (and dispatch focus)
+            onSearch(searchInput);
+        } else if (searchInput !== "") {
+            if (timeoutId !== null) clearTimeout(timeoutId);
+
+            // Wait a certain amount of time and start search, if no new key is pressed
+            timeoutId = setTimeout(() => onSearch(searchInput), timeout);
+        }
+    };
+</script>
+
+<!--
+@component
+Search bar with adaptable placeholder text.
+
+It starts the search on the following events:
+  - the user pressed enter
+  - the user click the search icon
+  - the user enters the search text and wait
+If the user enters the 'Esc' key, the search is aborted.
+
+Usage:
+```tsx
+    <SearchBar placeholderText={yourText} onSearch={yourCallback    } />
```

Comment:
Let's keep the description in one tense and use the svelte highlighting (tsx is for react + typescript).
```suggestion
It starts the search on the following events:
  - the user presses enter
  - the user clicks the search icon
  - the user enters the search text and waits
If the user enters the 'Esc' key, the search is aborted.

Usage:
```svelte
    <SearchBar placeholderText={yourText} onSearch={yourCallback} />
```

Reply:
I changed it

Reply:
There's still the gigantic whitespace after `yourCallback`.